src/content_mapblock.cpp

```diff
@@ -1000,7 +1012,10 @@ void MapblockMeshGenerator::drawPlantlikeRootedNode()
 	useTile(0, MATERIAL_FLAG_CRACK_OVERLAY, 0, true);
 	origin += v3f(0.0, BS, 0.0);
 	p.Y++;
-	if (data->m_smooth_lighting) {
+	bool fullbright = g_settings->getBool("fullbright_mode");
```

Comment:
This setting getter is too slow to use for every node drawn, fullbright mode needs to be cached in a bool.